### ISSUE TITLE
Sniper pretty printer doesn't handle for loops

### ISSUE THREAD

**Issue by oowekyala** [INRIA/spoon#3070]
First, thanks for this awesome project!

I ran into trouble when using the sniper pretty printer to print a class in which there is a for loop.

I put together [a gist as an MWE](https://gist.github.com/oowekyala/a1a77b2e271c2e7574dcb03dddad311c), the full stack traces are also included there.

When the loop has an unbraced single-statement body, the exception is **spoon.SpoonException: The element of class class spoon.support.reflect.code.CtForImpl does not have CtRole.STATEMENT**

<details>
 <summary>Stack trace</summary>

```
Exception in thread "main" spoon.SpoonException: The element of class class spoon.support.reflect.code.CtForImpl does not have CtRole.STATEMENT
	at spoon.reflect.meta.impl.RoleHandlerHelper.getRoleHandler(RoleHandlerHelper.java:55)
	at spoon.support.sniper.internal.ElementSourceFragment.getRoleHandler(ElementSourceFragment.java:241)
	at spoon.support.sniper.internal.ElementSourceFragment.addChild(ElementSourceFragment.java:181)
	at spoon.support.sniper.internal.ElementSourceFragment.access$000(ElementSourceFragment.java:44)
	at spoon.support.sniper.internal.ElementSourceFragment$1.enter(ElementSourceFragment.java:145)
	at spoon.reflect.visitor.CtScanner.visitCtAssignment(CtScanner.java:270)
	at spoon.support.reflect.code.CtAssignmentImpl.accept(CtAssignmentImpl.java:43)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:143)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:119)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:104)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:81)
	at spoon.reflect.visitor.CtScanner.visitCtBlock(CtScanner.java:294)
	at spoon.support.reflect.code.CtBlockImpl.accept(CtBlockImpl.java:55)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:143)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:119)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:104)
	at spoon.reflect.visitor.CtScanner.visitCtFor(CtScanner.java:458)
	at spoon.support.reflect.code.CtForImpl.accept(CtForImpl.java:38)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:143)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:119)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:104)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:81)
	at spoon.reflect.visitor.CtScanner.visitCtBlock(CtScanner.java:294)
	at spoon.support.reflect.code.CtBlockImpl.accept(CtBlockImpl.java:55)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:143)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:119)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:104)
	at spoon.reflect.visitor.CtScanner.visitCtAnonymousExecutable(CtScanner.java:221)
	at spoon.support.reflect.declaration.CtAnonymousExecutableImpl.accept(CtAnonymousExecutableImpl.java:35)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:143)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:119)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:104)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:81)
	at spoon.reflect.visitor.CtScanner.visitCtClass(CtScanner.java:330)
	at spoon.support.reflect.declaration.CtClassImpl.accept(CtClassImpl.java:56)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:143)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:119)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:104)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:81)
	at spoon.reflect.visitor.EarlyTerminatingScanner.visitCtCompilationUnit(EarlyTerminatingScanner.java:158)
	at spoon.support.reflect.declaration.CtCompilationUnitImpl.accept(CtCompilationUnitImpl.java:406)
	at spoon.reflect.visitor.EarlyTerminatingScanner.doScan(EarlyTerminatingScanner.java:143)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:119)
	at spoon.reflect.visitor.CtScanner.scan(CtScanner.java:165)
	at spoon.reflect.visitor.EarlyTerminatingScanner.scan(EarlyTerminatingScanner.java:104)
	at spoon.support.sniper.internal.ElementSourceFragment.addTreeOfSourceFragmentsOfElement(ElementSourceFragment.java:169)
	at spoon.support.reflect.declaration.CtCompilationUnitImpl.getOriginalSourceFragment(CtCompilationUnitImpl.java:358)
	at spoon.support.sniper.SniperJavaPrettyPrinter.printTypes(SniperJavaPrettyPrinter.java:145)
	at spoon.reflect.visitor.DefaultJavaPrettyPrinter.calculate(DefaultJavaPrettyPrinter.java:1984)
	at spoon.support.sniper.SniperJavaPrettyPrinter.calculate(SniperJavaPrettyPrinter.java:105)
	at spoon.support.JavaOutputProcessor.createJavaFile(JavaOutputProcessor.java:114)
	at MWE.main(App.java:22)
```
</details>

When the loop has a braced body, an NPE is thrown

<details>
 <summary>Stack trace</summary>

```
Exception in thread "main" java.lang.NullPointerException
	at spoon.support.sniper.internal.ElementSourceFragment.forEachConstantFragment(ElementSourceFragment.java:664)
	at spoon.support.sniper.internal.ElementSourceFragment.getChildrenFragments(ElementSourceFragment.java:464)
	at spoon.support.sniper.SniperJavaPrettyPrinter.printTypes(SniperJavaPrettyPrinter.java:146)
	at spoon.reflect.visitor.DefaultJavaPrettyPrinter.calculate(DefaultJavaPrettyPrinter.java:1984)
	at spoon.support.sniper.SniperJavaPrettyPrinter.calculate(SniperJavaPrettyPrinter.java:105)
	at spoon.support.JavaOutputProcessor.createJavaFile(JavaOutputProcessor.java:114)
```
</details>

I'm not 100% sure I'm using the Spoon API as intended, yet this doesn't happen when using the default pretty printer.

Also I didn't test that with other kinds of loops, the problem may be present as well. This problem appears very similar to #2838

**Comment by monperrus** [comment]
Thanks for the deduplication. Closing this one. @oowekyala please subscribe to #2938.

**Comment by nharrand** [comment]
Will be fixed when #3084 is merged.
